growthbook/growthbook
# GB-265 import from LD

### Features and Changes

Adds support for providing your API key and importing your projects, environments and features (basic) from LD.

Adds a new endpoint to create environments.


### Testing

- create a LD account
- create projects
- create environments
- create features in all the projects
- create an API key
- visit http://localhost:3000/importing and input your API key and see the import happen. you can also get there from the settings page, or the settings menu.

### Screenshots

#### Import your data landing page

An intermediary page that will show all competitors we can import from.

For now, it's just LD, but will include others as we build support for them. The logos and code is already set up and ready to go in the related Storybook story, we just need to copy/paste it over when we're ready to add it.

Currently:

<img width="1658" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/8838ecb3-ca55-495d-88ba-61ca8e57ea39">

Each service we import from will have its own page with breadcrumbs (note that some of the screenshots below do not have breadcrumbs because this was added after the screenshot was taken):

<img width="1018" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/92bb61f5-9398-4b62-942a-d8304edafa92">


The future:

<img width="1660" alt="Screen Shot 2023-07-06 at 2 03 42 PM" src="https://github.com/growthbook/growthbook/assets/113377031/cf48a567-d621-4e8d-a96f-ccae2fec3eb0">


<img width="1661" alt="Screen Shot 2023-07-06 at 2 03 30 PM" src="https://github.com/growthbook/growthbook/assets/113377031/b638da99-ffc0-4bd4-ad0d-c1bbef84b0d2">




#### Import from LD page

<img width="1078" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/0ef04b35-b427-4fcc-aacf-450cf55cdca6">


<img width="1315" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/14d8b47b-8961-4984-b90b-df4d26f76524">

Update: A grey dash shows for duplicate entries. 

<img width="270" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/46ac8bf8-5906-4690-99a0-b919b3d13ce9">
<img width="108" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/c1dfa63c-79ba-45b2-8ae7-c52353538216">
<img width="102" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/a810052b-c1db-4543-b208-0bb7a8ebc368">


Update: The red X's below only show for other errors.

![CleanShot 2023-06-26 at 17 09 26@2x](https://github.com/growthbook/growthbook/assets/113377031/f783dccb-f4e9-4062-9401-7d2ff6b94f8d)
